configure.ac

```diff
@@ -168,6 +168,12 @@ AX_BOOST_BASE([1.35],
 CPPFLAGS="$BOOST_CPPFLAGS $CPPFLAGS"
 LDFLAGS="$BOOST_LDFLAGS $LDFLAGS"
 
+# add workaround for problematic boost version
+AC_LANG_PUSH(C++)
+AC_COMPILE_IFELSE([_AX_BOOST_BASE_PROGRAM(106000)], [],
```

Comment:
`_AX_BOOST_BASE_PROGRAM` seems to be internal/undocumented. What's wrong with using `AX_BOOST_BASE()` ?

Reply:
>What's wrong with using AX_BOOST_BASE() ?

I tried that before this version. AX_BOOST_BASE will cause the `./configure` error out if the version requirement is not satisfied, it cannot be used for version detection.

>_AX_BOOST_BASE_PROGRAM seems to be internal/undocumented.

Yes, it comes from `AX_BOOST_BASE` module, it should be internal, but it is visible from the outside, I consider it more concise than writing our own program.

Reply:
@sledgehammer999 
Have second thoughts?